### PR TITLE
Added entry for Creational Design Patterns

### PR DESCRIPTION
**Pull request by ChowdhuryShaswata**
- Added a markdown file detailing the 5 most commonly used creational design patterns, including the usage, structure and benefits of each.
- Added self-created UML diagrams expressing the structure of these design patterns.
- Added links to further resources.

### CODE REVIEW

File: Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+<p align="center">
+<img width="690" alt="UML of Builder design pattern" src="../assets/Builder.png">
+</p>
+
+#### Benefits
+
+- Simplifies the creation process of complicated objects.
+- Allows the adjustment of individual steps without having to rewrite the entire creation process.
+
```
Comment (paridhi26) [inline Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md:116]: As a suggestion, adding a conclusion section summarizing each of the design patterns and comparing/contrasting them for particular scenarios that arise when programming would be a good wrap-up.

File: Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+## Further Reading <a name="further-reading"></a>
+
+Refactoring Guru provides a detailed reference guide with examples for creational design patterns, as well as other design patterns that can be useful to implement in the right scenarios.
```
Comment (paridhi26) [inline Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md:119]: Good job including extra resources. These links are helpful!

File: Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+- By using an abstract factory, the user is able to avoid having to implement separate code for each different object factory necessary for their product.
+- As with the factory method, the object creation system becomes highly extensible: if a new extension of an object is declared, the user can simply create another implementation of the abstract factory to produce it, without having to change the behavior of any other classes/functions that utilise the factory.
+
+### Singleton
+
+#### Usage
+
+A class or resource needs to only have one central instance that is globally accessed across the program.
+
+A singleton achieves 2 purposes: it maintains exactly 1 instance of a given class  across the entire program, and it provides a global access point to that class. It achieves this by first hiding the default constructor of the original class, and then providing a static constructor for the class. The first time the constructor is called, it generates an instance of the class and caches it within a static variable. Further calls to the constructor return the cached instance rather than creating a new one.
```
Comment (paridhi26) [inline Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md:80]: Converting the 2 purposes into a list of bullet points may improve readability.

File: Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+The factory method design pattern introduces a class containing a singular create method that creates one class of objects (products). Extensions of the factory class are created as needed to override the default implementation (or abstraction) of the create method to produce different subclasses of the product.
+
+#### Structure
+
+A factory class containing one (abstract) create method that produces a product class of the required type. Implementations of the factory override the create method as required to output subclasses of the product.
+
+<p align="center">
+<img width="690" alt="UML of Factory Method design pattern" src="../assets/Factory Method.png">
+</p>
+
```
Comment (paridhi26) [inline Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md:46]: Great job including the UML diagrams. Very instructive.

File: Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+#### Benefits
+
+- Allows variations of the product class to be easily created without having to define new calling behaviour in programs utilising the product and factory classes.
+
```
Comment (paridhi26) [inline Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md:50]: Optionally, I think adding short examples for each design pattern to illustrate what scenarios they could be used in could be a good addition! 

For example: When programming functionality for a bakery, a Simple Factory design pattern can be used to create different types of pastries such as croissants, muffins, and cookies. The factory takes input from customers (e.g., pastry type) and produces the corresponding pastry without exposing the complexities of the baking process.

File: Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
+# Creational Design Patterns
+
```
Comment (paridhi26) [inline Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md:2]: Adding a table of contents for easy navigation and overview of content on this page could be a good addition.

**Review by paridhi26 — CHANGES_REQUESTED** [review]
Good job explaining different creational design patterns! The usage of UML diagrams is very helpful. The document flows well and is easy to read. I have suggested some formatting improvements and additions that can further improve the readability of the document in the comments :)

File: Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
+# Creational Design Patterns
```
Comment (Aiforall123) [inline Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md:1]: I would try to add a table of content which would be good to jump from section to section.

**Review by Aiforall123 — CHANGES_REQUESTED** [review]
Excellent Shaswata! UML diagrams are very good and the explanation for each design pattern is good. Just try to add a table of content and conclusion :)

**Review by Aiforall123 — CHANGES_REQUESTED** [review]
Excellent Shaswata for adding the Conclusion and Table of Content. However, I cannot see the image being appeared correctly.

**Review by paridhi26 — APPROVED** [review]
Great job adding the table of contents and examples! The document is now easier to navigate thanks to the table of contents and the examples make the design patterns must easier to understand.

**Review by Aiforall123 — APPROVED** [review]
All the conflicts are now resolved and the images are appearing correctly. Ready to merge.

**Pull request merged**